aeron-system-tests/src/test/java/io/aeron/SessionSpecificSubscriptionTest.java

```diff
@@ -74,15 +77,23 @@ public void shouldSubscribeToSpecificSessionIdsAndWildcard()
         try (Subscription subscription1 = aeron.addSubscription(channelUriWithSessionId1, STREAM_ID);
             Subscription subscription2 = aeron.addSubscription(channelUriWithSessionId2, STREAM_ID);
             Subscription subscriptionWildcard = aeron.addSubscription(channelUriWithoutSessionId, STREAM_ID);
-            ExclusivePublication publication1 = aeron.addExclusivePublication(channelUriWithSessionId1, STREAM_ID);
-            ExclusivePublication publication2 = aeron.addExclusivePublication(channelUriWithSessionId2, STREAM_ID))
+            Publication publication1 = aeron.addExclusivePublication(channelUriWithSessionId1, STREAM_ID);
+            Publication publication2 = aeron.addExclusivePublication(channelUriWithSessionId2, STREAM_ID))
         {
             while (!publication1.isConnected() || !publication2.isConnected())
             {
                 SystemTest.checkInterruptedStatus();
                 Thread.yield();
             }
 
+            while (subscription1.imageCount() != 1 ||
+                    subscription2.imageCount() != 1 ||
+                    subscriptionWildcard.imageCount() != 2)
```

Comment:
we can't assert this conditionals here. Right? 
Because a publication is connected if there is at least one subscription but here it might be either `subscription1(2)` or `subscriptionWildcard`

Reply:
We cannot assert because we must wait until they all reach that state.